__device-tests__/gutenberg-editor-sanity-test-4-visual.test.js

```diff
@@ -0,0 +1,235 @@
+			await selectTextFromTextInput(
+				editorPage.driver,
+				buttonBlockTextInput
+			);
+
+			// Visual test check
+			const screenshot = await takeScreenshot( {
+				withoutKeyboard: true,
+			} );
+			expect( screenshot ).toMatchImageSnapshot();
+
+			await buttonBlockTextInput.click();
+
+			await editorPage.removeBlockAtPosition( blockNames.button );
+		} );
+
+		it( 'Edit text styles', async () => {
+			await editorPage.addNewBlock( blockNames.buttons );
+
+			const buttonsBlock = await editorPage.getBlockAtPosition(
+				blockNames.buttons
+			);
+
+			const firstButtonTextInput = await editorPage.getButtonBlockTextInputAtPosition();
+			await editorPage.typeTextToTextBlock(
+				firstButtonTextInput,
+				e2eTestData.listItem2
+			);
+			await selectTextFromTextInput(
+				editorPage.driver,
+				firstButtonTextInput
+			);
+
+			// Toggle Bold formatting for the first Button block
+			await editorPage.toggleFormatting( 'Bold' );
+
+			// Add a second button block
+			await editorPage.addButtonWithInlineAppender( 2 );
+			const secondButtonTextInput = await editorPage.getButtonBlockTextInputAtPosition(
+				2
+			);
+			await editorPage.typeTextToTextBlock(
+				secondButtonTextInput,
+				e2eTestData.listItem2
+			);
+			await selectTextFromTextInput(
+				editorPage.driver,
+				secondButtonTextInput
+			);
+
+			// Toggle Italic formatting for the first Button block
+			await editorPage.toggleFormatting( 'Italic' );
+
+			// Add a third button block
+			await editorPage.addButtonWithInlineAppender( 3 );
+			const thirdButtonTextInput = await editorPage.getButtonBlockTextInputAtPosition(
+				3
+			);
+			await editorPage.typeTextToTextBlock(
+				thirdButtonTextInput,
+				e2eTestData.listItem2
+			);
+			await selectTextFromTextInput(
+				editorPage.driver,
+				thirdButtonTextInput
+			);
+
+			// Toggle Strikethrough formatting for the first Button block
+			await editorPage.toggleFormatting( 'Strikethrough' );
+
+			// Unfocus the Buttons block
+			const titleElement = await editorPage.getTitleElement( {
+				autoscroll: true,
+			} );
+			await titleElement.click();
+
+			await editorPage.dismissKeyboard();
+
+			// Visual test check for unselected state
+			const screenshot = await takeScreenshot();
+			expect( screenshot ).toMatchImageSnapshot();
+
+			await buttonsBlock.click();
+			await editorPage.removeBlockAtPosition( blockNames.buttons );
+		} );
+
+		it( 'Link from the clipboard is presented as an option in the link picker', async () => {
+			// Increase threshold to avoid issues with the cursor position and keyboard.
+			const snapshotConfig = { failureThreshold: 0.03 };
+			await editorPage.addNewBlock( blockNames.buttons );
+
+			const link = 'https://wordpress.org/';
+
+			await setClipboard( editorPage.driver, link );
+
+			await editorPage.toggleFormatting( 'Edit link' );
+			await editorPage.openLinkToSettings();
+
+			// Wait for the modal to open
+			await editorPage.driver.sleep( 3000 );
```

Comment:
Not necessarily required right now — maybe it is premature — but I wonder if we should abstract the modal animation times so that it is (1) clear what is occurring and (2) could be modified across tests as necessary. 

```js
const MODAL_OPEN_DURATION = 3000;
const MODAL_CLOSE_DURATION = 2000;
```

Reply:
Good idea, maybe even having a `waitForModalToOpen` / `waitForModalToClose` utils. I can create a follow-up PR to introduce either that or just the constants.

Reply:
> My hot take is that we have a lot of test helpers, to the quantity that I find it difficult to comprehend each of them and their intent.

I share the same feeling 😅 

> I wonder if we would benefit from evaluating if there any helpers we might should remove and replace with composing other existing, lower-level helpers. My perception is that we have some helpers that are very specific and niche. Obviously, a subject to discuss in a different place. 

I agree, I think it's good you shared this, working on adding new tests always brings new perspectives and all the feedback will improve our current workflows and the way we do things.